content/300-guides/050-database/900-turso.mdx

```diff
@@ -0,0 +1,183 @@
+
+## Differences to consider
+
+libSQL is 100% compatible with SQLite. libSQL extends SQLite with the following features and capabilities:
+
+- Support for replication
+- Support for automated backups
+- Ability to embed Turso as part of other programs such as the Linux kernel
+- Supports user-defined functions
+- Support for asynchronous I/O
+
+To learn more about the differences between libSQL and how it is different from SQLite, see [libSQL Manifesto](https://turso.tech/libsql-manifesto).
+
+There are a number of differences between Turso and SQLite to consider. You should be aware of the following when deciding to use Turso and Prisma:
+
+- **Remote and embedded SQLite databases**. libSQL uses HTTP to connect to the remote SQLite database. libSQL also supports remote database replicas and embedded replicas. Embedded replicas enable you to replicate your primary database inside your application.
+- **Making schema changes**. Since libSQL uses HTTP to connect to the remote database, this makes it incompatible with Prisma Migrate. However, you can use [`prisma migrate diff`](/reference/api-reference/command-reference#migrate-diff) <span class="api"></span> to create a schema migration and then apply the changes to your database using [Turso's CLI](https://docs.turso.tech/reference/turso-cli).
+
+## Connect and query your primary database
+
+1. Create a database on Turso if you don't have one already:
+
+   ```sh
+   turso db create turso-prisma-db
+   ```
+
+2. Retrieve the database's connection string:
+
+   ```sh
+   turso db show turso-prisma-db
+   ```
+
+3. Create an authentication token that will allow you to connect to the database:
+
+   ```sh
+   turso db tokens create turso-prisma-db
+   ```
+
+4. Update your `.env` file with the authentication token and connection string:
+
+   ```text
+   TURSO_AUTH_TOKEN="eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9..."
+   TURSO_DATABASE_URL="libsql://turso-prisma-db-user.turso.io"
+   ```
+
+5. Enable the `driverAdapters` Preview feature flag:
+
+   ```prisma highlight=3;add
+   generator client {
+     provider        = "prisma-client-js"
+     previewFeatures = ["driverAdapters"]
+   }
+   
+   datasource db {
+     provider = "sqlite"
+     url      = "file:./dev.db"
+   }
+   ```
+
+6. Generate Prisma Client:
+
+   ```sh
+   npx prisma generate
+   ```
+
+7. Install the libSQL database client and Prisma driver adapter for libSQL packages:
+
+   ```sh
+   npm install @libsql/client @prisma/adapter-libsql
+   ```
+
+8. Update your Prisma Client instance with the following snippet:
+
+   ```ts
+   import { PrismaClient } from '@prisma/client'
+   import { PrismaLibSQL } from '@prisma/adapter-libsql'
+   import { createClient } from '@libsql/client'
+
+   const libsql = createClient({
+     url: `${process.env.TURSO_DATABASE_URL}`,
+     authToken: `${process.env.TURSO_AUTH_TOKEN}`,
+   })
+
+   const adapter = new PrismaLibSQL(libsql)
+   const prisma = new PrismaClient({ adapter })
+   ```
```

Comment:
Less scary list of instructions by splitting them:
````suggestion
Start by preparing your database on Turso:

1. Create a database on Turso if you don't have one already:

   ```sh
   turso db create turso-prisma-db
   ```

2. Retrieve the database's connection string:

   ```sh
   turso db show turso-prisma-db
   ```

3. Create an authentication token that will allow you to connect to the database:

   ```sh
   turso db tokens create turso-prisma-db
   ```

4. Update your `.env` file with the authentication token and connection string:

   ```text
   TURSO_AUTH_TOKEN="eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9..."
   TURSO_DATABASE_URL="libsql://turso-prisma-db-user.turso.io"
   ```

Now you can configure Prisma to use the Turso database:

1. Enable the `driverAdapters` Preview feature flag:

   ```prisma highlight=3;add
   generator client {
     provider        = "prisma-client-js"
     previewFeatures = ["driverAdapters"]
   }
   
   datasource db {
     provider = "sqlite"
     url      = "file:./dev.db"
   }
   ```

2. Generate Prisma Client:

   ```sh
   npx prisma generate
   ```

3. Install the libSQL database client and Prisma driver adapter for libSQL packages:

   ```sh
   npm install @libsql/client @prisma/adapter-libsql
   ```

4. Update your Prisma Client instance with the following snippet:

   ```ts
   import { PrismaClient } from '@prisma/client'
   import { PrismaLibSQL } from '@prisma/adapter-libsql'
   import { createClient } from '@libsql/client'

   const libsql = createClient({
     url: `${process.env.TURSO_DATABASE_URL}`,
     authToken: `${process.env.TURSO_AUTH_TOKEN}`,
   })

   const adapter = new PrismaLibSQL(libsql)
   const prisma = new PrismaClient({ adapter })
   ```
````

Reply:
I think we can remove the entire section on provisioning a database to keep it focused on connecting to the database only with the driver adapter 🤔